runtime/sema/type.go

```diff
@@ -636,6 +643,13 @@ func (t *OptionalType) Resolve(typeArguments *TypeParameterTypeOrderedMap) Type
 	}
 }
 
+func (t *OptionalType) SupportedEntitlements() *EntitlementOrderedSet {
+	if entitlementSupportingType, ok := t.Type.(EntitlementSupportingType); ok {
+		return entitlementSupportingType.SupportedEntitlements()
+	}
+	return orderedmap.New[EntitlementOrderedSet](0)
```

Comment:
Avoid allocating a new empty set

```suggestion
	return nil
```